st2common/bin/st2-run-pack-tests

```diff
@@ -194,6 +195,13 @@ if [ "${CREATE_VIRTUALENV}" = true ]; then
         mkdir -p ${VIRTUALENVS_DIR}
         virtualenv --no-download --system-site-packages ${VIRTUALENV_DIR}
 
+        if [ -f "${STACKSTORM_VIRTUALENV_PYTHON_BINARY}" ]; then
+            # ensure any .pth files in st2 venv get loaded with the pack venv too.
+            ST2_SITE_PACKAGES=$(${STACKSTORM_VIRTUALENV_PYTHON_BINARY} -c "import sysconfig;print(sysconfig.get_path('platlib'))")
+            PACK_SITE_PACKAGES=$(${VIRTUALENV_DIR}/bin/python3 -c "import sysconfig;print(sysconfig.get_path('platlib'))")
+            echo "import sys; addsitedir('${ST2_SITE_PACKAGES}', known_paths)" > "${PACK_SITE_PACKAGES}/zzzzzzzzzz__st2__.pth"
```

Comment:
extract a PR that adds support for pth files in the st2 venv (typically installed at `/opt/stackstorm/st2`). Pants uses PEP 660 wheels to do an editable install of the st2 sources, but that uses `.pth` files instead of symlinks to the sources. So, we need to make sure the pack venvs can still find the st2 sources in the main st2 venv.

This includes changes in:
- st2common/st2common/util/virtualenvs.py
- st2common/st2common/util/sandboxing.py
- st2common/tests/unit/test_util_sandboxing.py
- st2reactor/tests/integration/test_sensor_container.py
- st2common/bin/st2-run-pack-tests

Reply:
#6183